ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -1092,13 +1131,67 @@ export const DatasetPage = fp.flow(
           conceptSetId: null,
           domain: PREPACKAGED_DOMAINS[prepackaged],
         });
+        // check surveys
+        if (prepackaged === PrePackagedConceptSetEnum.SURVEY) {
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYBASICS);
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYLIFESTYLE);
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYOVERALLHEALTH);
+          updatedPrepackaged.add(
+            PrePackagedConceptSetEnum.SURVEYHEALTHCAREACCESSUTILIZATION
+          );
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYCOPE);
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYSDOH);
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYCOVIDVACCINE);
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYPFHH);
+        } else if (
+          prepackaged !== PrePackagedConceptSetEnum.SURVEY &&
+          Object.keys(PREPACKAGED_SURVEY_DOMAINS).every((key) =>
+            updatedPrepackaged.has(key.valueOf())
+          )
+        ) {
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEY);
+        }
       } else {
+        // check if unselected is survey
+        if (prepackaged === PrePackagedConceptSetEnum.SURVEY) {
+          updatedPrepackaged.delete(PrePackagedConceptSetEnum.SURVEYBASICS);
+          updatedPrepackaged.delete(PrePackagedConceptSetEnum.SURVEYLIFESTYLE);
+          updatedPrepackaged.delete(
+            PrePackagedConceptSetEnum.SURVEYOVERALLHEALTH
+          );
+          updatedPrepackaged.delete(
+            PrePackagedConceptSetEnum.SURVEYHEALTHCAREACCESSUTILIZATION
+          );
+          updatedPrepackaged.delete(PrePackagedConceptSetEnum.SURVEYCOPE);
+          updatedPrepackaged.delete(PrePackagedConceptSetEnum.SURVEYSDOH);
+          updatedPrepackaged.delete(
+            PrePackagedConceptSetEnum.SURVEYCOVIDVACCINE
+          );
+          updatedPrepackaged.delete(PrePackagedConceptSetEnum.SURVEYPFHH);
```

Comment:
Same as above for the deletes here:
```suggestion
          Object.keys(PREPACKAGED_SURVEY_DOMAINS).forEach((prepackagedSurvey) =>
            updatedPrepackaged.delete(prepackagedSurvey)
          );
```